### DISCUSSION TITLE
Spark on K8S(mode=cluster) kyuubi 找不到 driver 地址

### DISCUSSION THREAD

**Discussion by fuhaiq** [apache/kyuubi#2747] [Q&A] [answered]
嗨～已经在 K8S 上部署了 Kyuubi 引擎使用 Spark on K8S，在 spark.submit.deployMode=client 下，一切工作正常。

但是在 spark.submit.deployMode=cluster 下，Driver 会以一个 Pod 单独启动起来，然后拉起对应的 Executor。此时 Kyuubi 是通过 Pod 的 hostname 来找 Driver 的，而在 K8S 里，Pod 之间不能直接通信，必须借助 Service

![ZqLcMEFyHZ](https://user-images.githubusercontent.com/5200755/170414261-5695f020-1baf-4961-839f-325f33bd0813.jpg)

![jdisE700O4](https://user-images.githubusercontent.com/5200755/170414360-65fdc86b-258b-4931-9f76-e8402411cf01.jpg)

此时，如果手动创建一个和 hostname 同名的 Service 指向这个 Driver，是可以正常工作的。

请问有什么解决方法么？

**Comment by pan3793** [comment, marked as answer]
Please try turn off `kyuubi.frontend.connection.url.use.hostname`

**Reply by preetsindhal** [reply]
Hi,
I have similar issue, driver is trying to connect to kyuubi pod,  i have set kyuubi.frontend.connection.url.use.hostname=false but didn't work.
![Selection_015](https://user-images.githubusercontent.com/11793702/192282085-e2056506-02ce-406e-8a4b-54fdc76991e1.png)

**Reply by pan3793** [reply]
Which version are you using?

**Reply by preetsindhal** [reply]
this, from logs
INFO org.apache.kyuubi.server.KyuubiServer: Version: 1.7.0-SNAPSHOT, Revision: , Branch: , Java: 1.8, Scala: 2.12, Spark: 3.2.2, Hadoop: 3.3.4, Hive: 3.1.3, Flink: 1.15.2, Trino: 363
